src/lightning/fabric/fabric.py

```diff
@@ -142,6 +142,7 @@ def __init__(
         self._loggers = loggers if isinstance(loggers, list) else [loggers]
         self._models_setup: int = 0
         self._launched: bool = False
+        self._backward_called: bool = False
```

Comment:
Shouldn't this be in the `_FabricModule`? This wouldn't work well with multiple models

Reply:
What would be an example of it not working with multiple models? I have tried to put it into the FabricModule first, but it was more complex and I want to avoid it if not needed.

Reply:
I mean if you wrap two models and call `fabric.backward()` with the loss associated to one but `loss.backward()` with the other. It will work if they run sequentially of course but I find it mildly worrying to share module-specific attributes in a different object (Fabric).

What's complex about it? You always have the module reference available no?

Reply:
`fabric.backward()` doesn't have the module reference available in the general case. You were probably thinking that `fabric.backward()` would be able to do `model._backward_called = True` and then the backward hooks would check this on the model instance, but that's not possible if `model=None`, regardless of one or multiple models.

I just added two more scenarios with multiple models to the test case. Please take a look at them. At the moment, can't find any edge cases that would be problematic.

Reply:
You could register a forward hook that sets it to True to the model